File: lib/feature/data/model/request/return_order_filter_request.dart
enum ReturnStatus {
  PENDING,
  APPROVED,
  REJECTED,
  SHIPPED,
  RECEIVED,
  REFUNDED,
  EXCHANGED,
}


class ReturnOrderFilterRequest {
  final ReturnStatus? status; 
  final int? userId;
  final int? orderId;
  final int? branchId;
  final DateTime? fromDate;
  final DateTime? toDate;
  final int page;
  final int size;
  final String sortBy;
  final String sortDirection;

  ReturnOrderFilterRequest({
    this.status,
    this.userId,
    this.orderId,
    this.branchId,
    this.fromDate,
    this.toDate,
    this.page = 0,
    this.size = 30,
    this.sortBy = 'createdAt',
    this.sortDirection = 'DESC',
  });

  Map<String, dynamic> toJson() {
    return {
      if (status != null) 'status': status.toString(),
      if (userId != null) 'userId': userId,
      if (orderId != null) 'orderId': orderId,
      if (branchId != null) 'branchId': branchId,
      if (fromDate != null) 'fromDate': _formatDate(fromDate!),
      if (toDate != null) 'toDate': _formatDate(toDate!),
      'page': page,
      'size': size,
      'sortBy': sortBy,
      'sortDirection': sortDirection,
    };
  }

  String _formatDate(DateTime date) {
    return "${date.year.toString().padLeft(4, '0')}-"
           "${date.month.toString().padLeft(2, '0')}-"
           "${date.day.toString().padLeft(2, '0')} "
           "${date.hour.toString().padLeft(2, '0')}:"
           "${date.minute.toString().padLeft(2, '0')}:"
           "${date.second.toString().padLeft(2, '0')}";
  }
}

File: src/main/java/com/market/MSA/requests/filters/ReturnOrderFilterRequest.java
package com.market.MSA.requests.filters;

import com.market.MSA.constants.ReturnStatus;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

/**
 * Request DTO for filtering return orders Supports filtering by status, user, order, branch, and
 * date range
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ReturnOrderFilterRequest {

  /** Filter by return order status */
  ReturnStatus status;

  /** Filter by user ID */
  Long userId;

  /** Filter by order ID */
  Long orderId;

  /** Filter by branch ID */
  Long branchId;

  /** Filter by creation date from (inclusive) */
  LocalDateTime fromDate;

  /** Filter by creation date to (inclusive) */
  LocalDateTime toDate;

  /** Page number for pagination (0-based) */
  @Builder.Default Integer page = 0;

  /** Page size for pagination */
  @Builder.Default Integer size = 10;

  /** Sort field for ordering results */
  @Builder.Default String sortBy = "createdAt";

  /** Sort direction (ASC or DESC) */
  @Builder.Default String sortDirection = "DESC";
}
